src/pocketmine/entity/Entity.php

```diff
@@ -760,7 +760,7 @@ public function setMaxHealth($amount){
 	}
 
 	public function canCollideWith(Entity $entity){
-		return !$this->justCreated and $entity !== $this;
+		return !$this->justCreated and $entity !== $this and !($entity instanceof Player and !$entity->isSpectator());
```

Comment:
Translating code to english:
when the entity (this) is not justCreated and the $entity isn't this entity and if the entity is not a instance of a player and is not a spectator the
So the entity cannot be a player wich is in spectator ....